examples/Huggingface_Transformers/Transformer_handler_generalized.py

```diff
@@ -225,18 +235,15 @@ def inference(self, input_batch):
             
         # Handling inference for text_generation.
         if self.setup_config["mode"] == "text_generation":
-            for input in input_ids_batch:
-                prompt_length = len(self.tokenizer.decode(input[0]))
-                # for each text example from the batch, there is one instance of generated text 
-                # generated text is added at the starting text example  
-                # the output has a max length and is decoded in text
-                outputs = self.model.generate(input, max_length=150, do_sample=True, top_p=0.95, top_k=60)
-                generated = self.tokenizer.decode(input[0]) + self.tokenizer.decode(outputs[0])[prompt_length + 1 :]
-                
-                print("Generated text: " + generated)
-                
-                inferences.append(generated)
-            logger.info("Generated text: '%s'", generated)
+            if self.setup_config["model_parallel"]:
+                input_ids_batch = input_ids_batch.to("cuda:0")
+            outputs = self.model.generate(input_ids_batch, max_length=50, do_sample=True, top_p=0.95, top_k=60)
+            for i, x in enumerate(outputs):
+                inferences.append(self.tokenizer.decode(outputs[i], skip_special_tokens=True))
```

Comment:
should decoding happen in postprocess?

Reply:
ideally yes, it follows the pattern from previous code as inference list is defined in the inference.